.github/CONTRIBUTING.md

```diff
@@ -15,13 +15,70 @@ To contribute to this repository, start by forking the [official repository] and
 * If your PR fixes a bug, please provide a step-by-step description of how to reproduce the bug.
 * If your PR addresses an existing issue, please reference that issue in the title or description.
 
-### Coding style guide
+## Contribution process
 
-**TODO: Write style guide**
+Magento maintains a public roadmap for this and other [Magento Research] repositories in each project's issue board.
 
-## Contribution process
+Any and all community participation in this backlog is encouraged and appreciated.
+Even foundational infrastructure stories are available for a generous developer to take on.
+
+To get started, look for issues tagged with the **[help wanted]** labels.
+These issues are ready for community ownership.
+
+**Note:**
+*We also accept unsolicited new issues/features and pull request, but priority is given to issues in our roadmap that community developers have been kind enough to take on.*
+
+### Claiming an issue on the roadmap
+
+If you are interested in taking ownership of a roadmap feature or issue, we ask that you go through the following process.
+This helps us organize and forecast the progress of our projects.
+
+#### Step 1: Add an issue comment
+
+Add a comment on an issue expressing your interest in taking ownership of it.
+Make sure your GitHub profile includes an email address so we can contact you privately.
+
+#### Step 2: Meet with a maintainer
+
+A maintainer will contact you and ask to set up a real-time meeting to discuss the issue you are interested in owning.
+This meeting can be in person, video chat, audio chat, or text chat.
+
+In general this meeting is brief but can vary with the complexity of an issue.
+For larger issues, we may schedule follow-up meetings.
+
+During this meeting, we provide you with any additional materials or resources you need to work on the issue.
+
+#### Step 3: Provide an estimate
+
+We ask that you provide us an estimate of how long it will take you to complete the issue.
+
+If you require more time to provide a time frame for completion, you are allowed to take up to five business days to think about it.
+
+If you can't get back to us by that time, we understand!
+As a community developer, you are helping us out in addition to your regular job.
+We will un-assign you from this issue, but please feel free to contribute to another issue.
+
+#### Step 4: Work on the issue
+
+After you provide a target date, the issue is now "in progress", and
```

Comment:
Replace "target date" with "estimate" in alignment with recent CommEng discussion. Thanks!